core/src/main/java/io/aiven/klaw/service/TopicOverviewService.java

```diff
@@ -171,81 +171,101 @@ private void updateAvroSchema(
+            } else {
+              hashMapSchemaObj =
+                  schemaObjects.get(Integer.parseInt(Objects.requireNonNull(schemaVersionSearch)));
+              schemaOfObj = (String) hashMapSchemaObj.get("schema");
+              schemaMap.put("isLatest", "false");
+              schemaMap.put("id", hashMapSchemaObj.get("id") + "");
+              schemaMap.put("compatibility", hashMapSchemaObj.get("compatibility") + "");
+              schemaMap.put("version", "" + schemaVersionSearch);
+
+              if (schemaObjects.size() > 1) {
+                int indexOfVersion = allVersionsList.indexOf(Integer.parseInt(schemaVersionSearch));
+                if (indexOfVersion + 1 == allVersionsList.size()) {
+                  schemaMap.put("showNext", "false");
+                  schemaMap.put("showPrev", "true");
+                  schemaMap.put("prevVersion", "" + allVersionsList.get(indexOfVersion - 1));
+                } else {
+                  schemaMap.put("showNext", "true");
+                  schemaMap.put("showPrev", "true");
+                  schemaMap.put("prevVersion", "" + allVersionsList.get(indexOfVersion - 1));
+                  schemaMap.put("nextVersion", "" + allVersionsList.get(indexOfVersion + 1));
+                }
+              }
             }
-          }
 
-          schemaMap.put("env", schemaEnv.getName());
-          dynamicObj = OBJECT_MAPPER.readValue(schemaOfObj, Object.class);
-          schemaOfObj = WRITER_WITH_DEFAULT_PRETTY_PRINTER.writeValueAsString(dynamicObj);
-          schemaMap.put("content", schemaOfObj);
+            schemaMap.put("env", schemaEnv.getName());
+            dynamicObj = OBJECT_MAPPER.readValue(schemaOfObj, Object.class);
+            schemaOfObj = WRITER_WITH_DEFAULT_PRETTY_PRINTER.writeValueAsString(dynamicObj);
+            schemaMap.put("content", schemaOfObj);
+
+            schemaDetails.add(schemaMap);
+            topicOverview.setSchemaExists(true);
 
-          schemaDetails.add(schemaMap);
-          topicOverview.setSchemaExists(true);
-          log.info("Getting schema " + topicNameSearch);
+            // Set Promotion Details
+            processSchemaPromotionDetails(topicOverview, tenantId, schemaEnv);
+            log.info("Getting schema details for: " + topicNameSearch);
+          }
         } catch (Exception e) {
           log.error("Error ", e);
         }
       }
+
       if (topicOverview.isSchemaExists()) topicOverview.setSchemaDetails(schemaDetails);
     }
   }
 
+  private void processSchemaPromotionDetails(
+      TopicOverview topicOverview, int tenantId, Env schemaEnv) {
+    log.info("SchemaEnv Id {}", schemaEnv.getId());
+    Map<String, String> promotionDetails = new HashMap<>();
+    generatePromotionDetails(
+        tenantId,
+        promotionDetails,
+        Arrays.asList(schemaEnv.getId()),
+        mailService.getEnvProperty(tenantId, "ORDER_OF_SCHEMA_ENVS"));
+    topicOverview.setPromotionDetails(promotionDetails);
```

Comment:
Question about field 'promotionDetails', if we are re-using it for schemas too ?

Reply:
It's a good question, 
Currently both Topics and Schemas use the same TopicOverview.java entry and both are on separate API calls.
But I could also see maybe something more like a TopicOveriew and SchemaOverview class with a common Overview parent allowing us to be a little more specific in the field names without having to keep adding fields to one Java class?